### PR TITLE
dhd: Fix android-config.h generating with busybox sed

### PR DESCRIPTION
**Pull request by Thaodan**

### CODE REVIEW

File: droid-hal-device.inc [hunk 1]
```diff
@@ -596,7 +596,8 @@ popd 1>/dev/null
 
 # Add our config into the default android-config.h
 echo Making new $HDRS/android-config.h
-sed '/CONFIG GOES HERE/,$d' $HDRS/android-config.h > $HDRS/android-config.h.new
+TOKEN_LINE=$(sed -n '/CONFIG GOES HERE/=' $HDRS/android-config.h)
```
Comment (sledges) [inline droid-hal-device.inc:599]: Uppercase variables are usually for shell's use, or conventionally for global exports. Script-internal (local) variables best look when in lowercase.

**Review by sledges — CHANGES_REQUESTED** [review]
Commit should read ```android-config.h``` I believe.

**Comment by Thaodan** [comment]
That is true for shell scripts but in this case all variables are mostly 
uppercase and wanted ti follow the rest of the specfile.

File: droid-hal-device.inc (hunk 1 above)
Comment (sledges) [reply droid-hal-device.inc:599]: > That is true for shell scripts but in this case all variables are mostly uppercase and wanted ti follow the rest of the specfile.

That doesn't mean we shouldn't strive to improve our coding style:)

**Comment by sledges** [comment]
> Commit should read `android-config.h` I believe.

As well as the next 3 commits do not seem to belong to this PR topic.

**Comment by sledges** [comment]
> As well as the next 2 commits do not seem to belong to this PR topic.

I have included those two useful commits in https://github.com/mer-hybris/droid-hal-device/pull/282

File: droid-hal-device.inc [hunk 2]
```diff
@@ -373,10 +373,10 @@ fi
 %if 0%{?_obs_build_project:1}
 
 # Set up kernel extra version for OBS kernel builds
-if EXTRAVERSION_OLD=$(cat %android_root/kernel/Makefile | grep "EXTRAVERSION ="); then
-    sed -i "s/$EXTRAVERSION_OLD/EXTRAVERSION = +%{version}/g" %android_root/kernel/Makefile
-elif EXTRAVERSION_OLD=$(cat %android_root/linux/kernel/Makefile | grep "EXTRAVERSION ="); then
-    sed -i "s/$EXTRAVERSION_OLD/EXTRAVERSION = +%{version}/g" %android_root/linux/kernel/Makefile
+if extra_version_old=$(grep"EXTRAVERSION =" %android_root/kernel/Makefile); then
```
Comment (sledges) [inline droid-hal-device.inc:376]: ```extra_version_old``` - no such variable referenced afterwards
```grep"EXTRAVERSION ="``` - space missing (will fail as command)
Have you tested this .spec in action?

**Review by sledges — CHANGES_REQUESTED** [review]

**Comment by sledges** [comment]
> Commit should read `android-config.h` I believe.

Commit still references a non-existing `droid-config.h` file.

File: droid-hal-device.inc (hunk 2 above)
Comment (Thaodan) [reply droid-hal-device.inc]: Yes I had after I fixed the typo.

File: droid-hal-device.inc [hunk 3]
```diff
@@ -373,10 +373,10 @@ fi
 %if 0%{?_obs_build_project:1}
 
 # Set up kernel extra version for OBS kernel builds
-if EXTRAVERSION_OLD=$(cat %android_root/kernel/Makefile | grep "EXTRAVERSION ="); then
-    sed -i "s/$EXTRAVERSION_OLD/EXTRAVERSION = +%{version}/g" %android_root/kernel/Makefile
-elif EXTRAVERSION_OLD=$(cat %android_root/linux/kernel/Makefile | grep "EXTRAVERSION ="); then
-    sed -i "s/$EXTRAVERSION_OLD/EXTRAVERSION = +%{version}/g" %android_root/linux/kernel/Makefile
+if extra_version_old=$(grep "EXTRAVERSION =" %android_root/kernel/Makefile); then
```
Comment (sledges) [inline droid-hal-device.inc:376]: This is still not fixed, should be `extraversion_old`.

**Review by sledges — CHANGES_REQUESTED** [review]

File: droid-hal-device.inc [hunk 4]
```diff
@@ -421,7 +421,7 @@ fi
 echo Building local tools
```
Comment (sledges) [inline droid-hal-device.inc:421]: There are 4 more occurrences of `test` still not converted to `[ ]`. 3 above and one way below.

**Review by sledges — CHANGES_REQUESTED** [review]

**Review by sledges — CHANGES_REQUESTED** [review]
Variables still in uppercase:
https://github.com/mer-hybris/droid-hal-device/blob/0a2b1ef02429c86a9bf81fd51fec6a3afc71d413/droid-hal-device.inc#L408
https://github.com/mer-hybris/droid-hal-device/blob/0a2b1ef02429c86a9bf81fd51fec6a3afc71d413/droid-hal-device.inc#L697

**Review by sledges — APPROVED** [review]

**Pull request merged**